{{cookiecutter.project_slug}}/server/{{cookiecutter.project_slug}}/settings.py

```diff
@@ -21,6 +13,13 @@ def _env_get_required(setting_name):
 IN_STAGING = ENVIRONMENT == "staging"
 IN_PROD = ENVIRONMENT == "production"
 
+def _env_get_required(setting_name):
+    """Get the value of an environment variable and assert that it is set."""
+    setting = os.environ.get(setting_name)
+    if IN_STAGING or IN_PROD:
+        assert setting not in {None, ""}, "{0} must be defined as an environment variable.".format(setting_name)
+    return setting
```

Comment:
Yea all of the env variables cause some problems for getting lightweight pytest going. But there is a better solution to that problem.
I think you make a good call out. Especially since it undermines what the function name says it's doing, which adds complexity and confusion.
Honestly I've also disliked the use of `assert` here, but have ignored it up until now since it technically happens outside of the running app, which is where assert can be a problem.
I believe there are libraries that make managing environment variables like this easier. I've used one or two in the pass and will do a quick look to see if there is a more thoughtful solution here.